geomesa-tools/src/main/scala/org/locationtech/geomesa/tools/accumulo/commands/EnvironmentCommand.scala

```diff
@@ -24,46 +26,78 @@ class EnvironmentCommand(parent: JCommander) extends Command(parent) with LazyLo
-    if (params.sfts == null && params.converters == null) {
-      // default - list all
-      listSfts()
+    if (params.sfts == null && params.converters == null && !params.listSfts &&
+      !params.listConverters && !params.describeSfts && !params.describeConverters) {
+      // default - print out parameters
       println
-      listConverters()
-    } else if (params.sfts != null) {
-      // only list specified
-      listSfts(params.sfts.toList)
+      parent.usage(command)
+    } else {
+      if (params.listSfts) {
+        listSftsNames()
+      }
+      if (params.listConverters){
+        listConverterNames()
+      }
+      if (params.describeSfts) {
+        listSfts()
+      }
+      if (params.describeConverters) {
+        listConverters()
+      }
+      if (params.sfts != null) {
+        listSfts(params.sfts.toList)
+      }
       if (params.converters != null) {
-        println
         listConverters(params.converters.toList)
       }
-    } else {
-      listConverters(params.converters.toList)
     }
   }
 
   def listSfts(names: List[String] = List.empty): Unit = {
     val all = SimpleFeatureTypeLoader.sfts
     val filtered = if (names.isEmpty) all else names.flatMap(n => all.find(_.getTypeName == n))
-    println("Simple Feature Types:")
+    println("\nSimple Feature Types:")
     if (filtered.isEmpty) {
       println("\tNone available")
     } else {
-      filtered.map(s => s"\t${s.getTypeName} = ${SimpleFeatureTypes.encodeType(s)}").foreach(println)
+      val paramsLower = params.format.toLowerCase
+      if(paramsLower == "typesafe" || paramsLower == "spec") {
+        val formatFn = paramsLower match {
+          case "typesafe" =>
+            (sft: SimpleFeatureType) => s"${SimpleFeatureTypes.toConfigString(sft, !params.excludeUserData, params.concise)}"
+          case "spec" =>
+            (sft: SimpleFeatureType) => s"${SimpleFeatureTypes.encodeType(sft, !params.excludeUserData)}"
+        }
+        filtered.sortBy(_.getTypeName).map(s => s"\t${s.getTypeName} = ${formatFn(s)}").foreach(println)
+      } else {
+        logger.error(s"Unknown config format: ${params.format}")
+      }
     }
   }
 
   def listConverters(names: List[String] = List.empty): Unit = {
     val all = ConverterConfigLoader.confs
     val filtered = if (names.isEmpty) all else names.flatMap(n => all.find(_._1 == n))
-    println("Simple Feature Type Converters:")
+    println("\nSimple Feature Type Converters:")
     if (filtered.isEmpty) {
       println("\tNone available")
     } else {
       val options = ConfigRenderOptions.defaults().setJson(false).setOriginComments(false)
       def render(c: Config) = c.root().render(options).replaceAll("\n", "\n\t")
-      filtered.map { case (name, conf)=> s"\tconverter-name=$name\n\t${render(conf)}\n"}.foreach(println)
+      filtered.map { case (name, conf)=> s"\tconverter-name=$name\n\t${render(conf)}\n"}.toArray.sortBy(_.self).foreach(println)
```

Comment:
i almost want to remove the tab here so that i can pipe it through some other command more easily